libraries/plugins/elasticsearch/include/graphene/elasticsearch/elasticsearch_plugin.hpp

```diff
@@ -174,9 +175,103 @@ struct bulk_struct {
    optional<visitor_struct> additional_data;
 };
 
+struct adaptor_struct {
+   variant adapt(const variant_object& op)
+   {
+      fc::mutable_variant_object o(op);
+      vector<string> keys_to_rename;
+      for (auto i = o.begin(); i != o.end(); ++i)
+      {
+         auto& element = (*i).value();
+         if (element.is_object())
+         {
+            const string& name = (*i).key();
+            auto& vo = element.get_object();
+            if (vo.contains(name.c_str()))
+               keys_to_rename.emplace_back(name);
+            element = adapt(vo);
+         }
+         else if (element.is_array())
+            adapt(element.get_array());
+      }
+      for (const auto& i : keys_to_rename)
+      {
+         string new_name = i + "_";
+         o[new_name] = variant(o[i]);
+         o.erase(i);
+      }
+
+      if (o.find("memo") != o.end())
+      {
+         auto& memo = o["memo"];
+         if (memo.is_string())
+         {
+            o["memo_"] = variant(o["memo"]);
+            o.erase("memo");
+         }
+         else if (memo.is_object())
+         {
+            fc::mutable_variant_object tmp(memo.get_object());
+            if (tmp.find("nonce") != tmp.end())
+            {
+               tmp["nonce"] = tmp["nonce"].as_string();
+               o["memo"] = tmp;
+            }
+         }
+      }
+      if (o.find("kind") != o.end())
+      {
+         o["kind"] = o["kind"].as_string();
+      }
+      if (o.find("new_parameters") != o.end())
```

Comment:
info: this is needed or ES will fail to insert with error:
`java.lang.IllegalArgumentException: Can't merge a non object mapping [operation_history.op_object.new_parameters.current_fees.parameters] with an object mapping [operation_history.op_object.new_parameters.current_fees.parameters]
`